velox/type/Type.cpp

```diff
@@ -1011,6 +1011,28 @@ std::string DateType::toString(int32_t days) const {
   return Timestamp::tmToString(tmValue, 0, options);
 }
 
+std::string DateType::toString(
```

Comment:
Let's remove this method and move the implementation into DateType::toIso8601.

Let's then update DateType::toString method to call DateType::toIso8601.

```
std::string DateType::toString(int32_t days) const {
   return DateType::toIso8601(days);
}
```

Reply:
The reason I introduced this separation was because you pointed out that we wanted to avoid modifying DateType::toString, to prevent disruption of the output format its existing callers may be dependent on.

Reply:
No caller should depend on the output of toString(). That output is strictly for logging and debugging. Hence, a function like to_iso1801 cannot use toString().

Reply:
Right, I understand that from our prior conversation - also another reason why I thought it best to separate the methods out, to avoid to_iso8601 from having to rely on toString(), while also not disrupting its format for any existing logging or casting operations that appear to be relying on it. Perhaps I misinterpreted something.

Modified to remove overload and have `toString()` call `toIso8601()`.